torch/onnx/symbolic_opset9.py

```diff
@@ -717,11 +714,15 @@ def _slice(g: jit_utils.GraphContext, input, axes, starts, ends):
 
 @_beartype.beartype
 def _maybe_cast_reduce_op_input(g: jit_utils.GraphContext, self):
-    dtype = self.type().scalarType()
-    # This check only covers traced modules where dtype is present
-    if dtype is not None:
+    scalar_type = _type_utils.JitScalarType.from_value(
+        self, _type_utils.JitScalarType.UNDEFINED
+    )
+    if scalar_type != _type_utils.JitScalarType.UNDEFINED:
```

Comment:
Whim: I wish python has Swift's implicit member expression, then this becomes `scalar_type != .UNDEFINED`